inst/BS5/assets/pkgdown.js

```diff
@@ -153,4 +153,136 @@ async function searchFuse(query, callback) {
+    <path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></svg>`,
+    "circle-half": `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" class="bi bi-circle-half" style="height:1em;width:1em;fill:currentColor;" aria-hidden="true" role="img" ><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></svg>`
+  }
+
+  const icons = {
+    "sun-fill": `<i class="bi bi-sun-fill"></i>`,
+    "moon-stars-fill": `<i class="bi bi-moon-stars-fill"></i>`,
+    "circle-half": `<i class="bi bi-circle-half"></i>`
+  }
+
+  return `<button class="btn btn-link nav-link py-2 px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (light)">
+    <span class="theme-icon-active">${icons['sun-fill']}</span>
+    <span class="d-lg-none ms-2" id="bd-theme-text">Toggle theme</span>
+  </button>
+  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
+    <li>
+      <button type="button" class="dropdown-item d-flex align-items-center active" data-bs-theme-value="light" aria-pressed="true">
+        <span class="me-2 opacity-50 theme-icon">${icons['sun-fill']}</span>
+        Light
+        <svg class="bi ms-auto d-none"><use href="#check2"></use></svg>
+      </button>
+    </li>
+    <li>
+      <button type="button" class="dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
+        <span class="me-2 opacity-50 theme-icon">${svgs['moon-stars-fill']}</span>
+        Dark
+        <svg class="bi ms-auto d-none"><use href="#check2"></use></svg>
+      </button>
+    </li>
+    <li>
+      <button type="button" class="dropdown-item d-flex align-items-center" data-bs-theme-value="auto" aria-pressed="false">
+        <span class="me-2 opacity-50 theme-icon">${icons['circle-half']}</span>
+        Auto
+        <svg class="bi ms-auto d-none"><use href="#check2"></use></svg>
+      </button>
+    </li>
+  </ul>`
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+  const switcher = document.querySelector('.navbar [href$="#dark-mode"]')
+  if (!switcher) return
+
+  switcher.parentElement.classList.add('dropdown')
+  switcher.parentElement.innerHTML = colorPickerMarkup()
+  bsSetupThemeToggle()
```

Comment:
If the markup were part of the page already, then this pattern of calling `bsSetupThemeToggle()` on document load would still work, you just wouldn't need to insert the dropdown markup. 

I do want to call out that it's important that `setTheme(getPreferredTheme())` is called as early as possible (so it can't be part of this event listener) so that the dark mode attribute is set before the content is shown. Otherwise, you'll end up with a flash of light mode styles.